asset-transfer-basic/rest-api-typescript/src/config.ts

```diff
@@ -0,0 +1,274 @@
+
+/*
+ * The port to start the REST server on
+ */
+export const port = env
+  .get('PORT')
+  .default('3000')
+  .example('3000')
+  .asPortNumber();
+
+/*
+ * The type of backoff to use for retrying failed submit jobs
+ */
+export const submitJobBackoffType = env
+  .get('SUBMIT_JOB_BACKOFF_TYPE')
+  .default('fixed')
+  .asEnum(['fixed', 'exponential']);
+
+/*
+ * Backoff delay for retrying failed submit jobs in milliseconds
+ */
+export const submitJobBackoffDelay = env
+  .get('SUBMIT_JOB_BACKOFF_DELAY')
+  .default('3000')
+  .example('3000')
+  .asIntPositive();
+
+/*
+ * The total number of attempts to try a submit job until it completes
+ */
+export const submitJobAttempts = env
+  .get('SUBMIT_JOB_ATTEMPTS')
+  .default('5')
+  .example('5')
+  .asIntPositive();
+
+/*
+ * The maximum number of submit jobs that can be processed in parallel
+ */
+export const submitJobConcurrency = env
+  .get('SUBMIT_JOB_CONCURRENCY')
+  .default('5')
+  .example('5')
+  .asIntPositive();
+
+/*
+ * The number of completed submit jobs to keep
+ */
+export const maxCompletedSubmitJobs = env
+  .get('MAX_COMPLETED_SUBMIT_JOBS')
+  .default('1000')
+  .example('1000')
+  .asIntPositive();
+
+/*
+ * The number of failed submit jobs to keep
+ */
+export const maxFailedSubmitJobs = env
+  .get('MAX_FAILED_SUBMIT_JOBS')
+  .default('1000')
+  .example('1000')
+  .asIntPositive();
+
+/*
+ * Whether to initialise a scheduler for the submit job queue
+ * There must be at least on queue scheduler to handle retries and you may want
+ * more than one for redundancy
+ */
+export const submitJobQueueScheduler = env
+  .get('SUBMIT_JOB_QUEUE_SCHEDULER')
+  .default('true')
+  .example('true')
+  .asBoolStrict();
+
+/*
+ * Whether to convert discovered host addresses to be 'localhost'
+ * This should be set to 'true' when running a docker composed fabric network on the
+ * local system, e.g. using the test network; otherwise should it should be 'false'
+ */
+export const asLocalhost = env
+  .get('AS_LOCAL_HOST')
+  .default('true')
+  .example('true')
+  .asBoolStrict();
+
+/*
+ * The Org1 MSP ID
+ */
+export const mspIdOrg1 = env
+  .get('HLF_MSP_ID_ORG1')
+  .default(`${ORG1}MSP`)
+  .example(`${ORG1}MSP`)
+  .asString();
+
+/*
+ * The Org2 MSP ID
+ */
+export const mspIdOrg2 = env
+  .get('HLF_MSP_ID_ORG2')
+  .default(`${ORG2}MSP`)
+  .example(`${ORG2}MSP`)
+  .asString();
+
+/*
+ * Name of the channel which the basic asset sample chaincode has been installed on
+ */
+export const channelName = env
+  .get('HLF_CHANNEL_NAME')
+  .default('mychannel')
+  .example('mychannel')
+  .asString();
+
+/*
+ * Name used to install the basic asset sample
+ */
+export const chaincodeName = env
+  .get('HLF_CHAINCODE_NAME')
+  .default('basic')
+  .example('basic')
+  .asString();
+
+/*
+ * The transaction submit timeout in seconds for commit notification to complete
+ */
+export const commitTimeout = env
+  .get('HLF_COMMIT_TIMEOUT')
+  .default('3000')
```

Comment:
3000 seconds, really?

Reply:
That does seem suspicious! I'll check!

Reply:
Updated to match the SDK default of 300 seconds
hyperledgendary/fabric-rest-sample#54